src/throttler-storage-redis.service.ts

```diff
@@ -25,21 +22,49 @@ export class ThrottlerStorageRedisService implements ThrottlerStorageRedis, OnMo
     }
   }
 
+  /**
+   * @description: This method look for members (records) in a Redis set, see: https://redis.io/docs/data-types/sets/
+   *
+   * @param key
+   */
   async getRecord(key: string): Promise<number[]> {
-    const ttls = (
-      await this.redis.scan(
-        0,
-        'MATCH',
-        `${this.redis?.options?.keyPrefix}${key}:*`,
-        'COUNT',
-        this.scanCount,
-      )
-    ).pop();
-    return (ttls as string[]).map((k) => parseInt(k.split(':').pop())).sort();
+    const setMembers = await this.redis.smembers(key);
+    const now = +new Date();
+
+    /*
+     That is a Redis SET cleaning mechanism.
+     - There is no way to set ttl for each set member separately.
+     - Each time new member is added to the set, set expiration is moving forward.
+     - We need to clean expired members manually (to avoid extra memory usage).
+     - That is needed in cases when set not expires during long period of time (because of continuous addRecord triggers)
+    */
```

Comment:
Your comment is very vague. You're using dashes on each line, which gives the indication that these are steps that we should do programmatically, which doesn't seem to be the case. 

I think you can change this whole block to simply this:

```suggestion
    // Clean expired members manually (to avoid extra memory usage)
```